### ISSUE TITLE
start.sh no more compatible with docker compose V2

### ISSUE THREAD

**Issue by HervePruvost** [theCalcaholic/ncp-monitoring-dashboard#3]
Hi,
since with V2 the "docker-compose" command is obsolete and has been replaced by "docker compose" (without hyphen), the install process with the start.sh script is broken.

I have changed locally the script to fit in with V2:  replaced COMPOSE_CMD="docker-compose" by COMPOSE_CMD="docker compose". 
But it is somehow not enough since the generated "./docker-compose" binary file, when installing docker compose over CURL, is not working, even making it executable with chmod -x.

For that reason I had to skip resp. comment out the systemd block in start.sh and avoid setting up the systemd service since the field "
ExecStartPre=${DOCKER_COMPOSE_CMD} pull
ExecStart=${DOCKER_COMPOSE_CMD} up"
in the ncp-monitoring-dashboard.service file are throwing "Exec Format Error" since "./docker-compose" binary is not working. 

I ended up running the rest of the start.sh script without compose curl installation and systemd setup. I reinstalled docker compose manually and started the dashboard directly using "docker compose up -d" from my local repository. Grafana is then running with all the other containers. But when visiting grafana the dashboard shows "No Data" in all panels.

My platform:
- RPI4
- Debian GNU/Linux 11
- VERSION="11 (bullseye)"
- architecture 64 bits

If you have an idea what is going wrong, that would be helpful. And the start.sh needs to be updated to fit with new docker compose versions.

Thx

**Comment by ADadlDudlDa** [comment]
Hello @HervePruvost,

I am getting the following error message after executing start.sh: "Job for ncp-monitoring-dashboard.service failed because the control process exited with error code."

Is this the same error you are getting?

How can I verify this?

Thank you.

ADadlDudlDa

**Comment by PositivelyCharged** [comment]
> Hi,
> 
> since with V2 the "docker-compose" command is obsolete and has been replaced by "docker compose" (without hyphen), the install process with the start.sh script is broken.
> 
> 
> 
> I have changed locally the script to fit in with V2:  replaced COMPOSE_CMD="docker-compose" by COMPOSE_CMD="docker compose". 
> 
> But it is somehow not enough since the generated "./docker-compose" binary file, when installing docker compose over CURL, is not working, even making it executable with chmod -x.
> 
> 
> 
> For that reason I had to skip resp. comment out the systemd block in start.sh and avoid setting up the systemd service since the field "
> 
> ExecStartPre=${DOCKER_COMPOSE_CMD} pull
> 
> ExecStart=${DOCKER_COMPOSE_CMD} up"
> 
> in the ncp-monitoring-dashboard.service file are throwing "Exec Format Error" since "./docker-compose" binary is not working. 
> 
> 
> 
> I ended up running the rest of the start.sh script without compose curl installation and systemd setup. I reinstalled docker compose manually and started the dashboard directly using "docker compose up -d" from my local repository. Grafana is then running with all the other containers. But when visiting grafana the dashboard shows "No Data" in all panels.
> 
> 
> 
> My platform:
> 
> - RPI4
> 
> - Debian GNU/Linux 11
> 
> - VERSION="11 (bullseye)"
> 
> - architecture 64 bits
> 
> 
> 
> If you have an idea what is going wrong, that would be helpful. And the start.sh needs to be updated to fit with new docker compose versions.
> 
> 
> 
> Thx

Yes, from my understanding the start.sh pulls the latest version of Grafana and it is Grafana that is incompatible as some of the handlers are now depreciated. The work around (if you can call it that) is to use an older version of Grafana and install the metrics script manually, rather than letting the script pull a Grafana update. I believe the latest version is 11.something and I was fine up to about 9.4..... I think.
